public/app/tasks/task.directive.js

```diff
@@ -28,8 +28,12 @@
                  toastr.error('Task ' + self.task.title + ' deleted!');
             }
 
+            function _debounceUpdateTaskTitle() { 
+                _.debounce(_updateTaskTitle(), 600); 
```

Comment:
Maybe I told you about its functionality a little tricky. Read documentation about debounce  using here - https://lodash.com/docs/4.16.4#debounce